src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -291,6 +339,41 @@ def reweight(self, svc_id, weight):
             id=int(svc_id),
             weight=float(weight))
 
+    @Endpoint('GET', version=APIVersion.EXPERIMENTAL)
+    @ReadPermission
+    @raise_if_no_orchestrator([OrchFeature.DAEMON_LIST])
+    @handle_orchestrator_error('host')
+    def deployment_options(self):
+        orch = OrchClient.instance()
+        hdds = 0
+        ssds = 0
+        nvmes = 0
+        res = copy.deepcopy(device_status_template)
+        devices = {}
+        for inventory_host in orch.inventory.list(hosts=None, refresh=True):
+            for device in inventory_host.devices.devices:
+                if device.available:
+                    devices[device.path] = device
+                    if device.human_readable_type == 'hdd':
+                        hdds += 1
+                    elif device.human_readable_type == 'ssd':
+                        ssds += 1
+                    elif device.human_readable_type == 'nvme':
+                        nvmes += 1
+        if hdds:
+            res['options']['cost-capacity']['available'] = True
+            res['recommended_option'] = 'cost-capacity'
+        return res
```

Comment:
Even if you don't implement the whole logic for this right now you should try to start modelling a bit how this should work:
* You need a 'logic' that you feed with inputs and outputs either a single value, being that a choice from: {capacity, throughput, iops}, or 3 values, a percentage of how each one is matching {99%, 50%, 0%}.
* So what's the structure of those inputs? I'd say those inputs are per-host, since we need, per host, the ratio of HDDs to SSDs/NVMes, [cpus and memory per OSD](https://docs.ceph.com/en/latest/start/hardware-recommendations/#minimum-hardware-recommendations).

```py
class HostFeatures:
  has_hdds: bool
  has_ssds: bool
  has_ nvmes: bool
  hdds_per_fast_device: float
  num_cores: int
  network_throughput: int
  ram: int
```
With the input of multiple `HostFeatures`, you should output either the optimal deployment, or some weights for each of them:
```py
>>> recommend_deployment([HostFeature1, HostFeature2, ...])
NamedTuple(capacity=0.3, throughput=0.9, nvme=0.0)
```

Reply:
https://tracker.ceph.com/issues/54553 added a tracker for this. I will move forward to push this PR ASAP and then work on that in another PR.